docs/tutorial/part04.rst

```diff
@@ -2,5 +2,168 @@
+
+..  todo::
+    Put screenshot image of the console here.
+
+Neomodel Query API
+==================
+Each of your models have some properties and methods(inherited from StructureNode or DjangoNode)
+that help us to express queries to the Neo4j database.
+
+The NodeSet object under the ``.nodes`` class property.
+--------------------------------------------------------
+A NodeSet represent a set of nodes matching common query parameters or filters.
+
+The ``.nodes`` property of each model store a NodeSet object. This NoseSet represent all the nodes mapped under a model.
+So ``Entity.nodes`` contains all the nodes with the label Entity on the database.
+If we wanted to count all the Entity nodes that are stored in the database, we just call the ``len`` python function
+over the ``Entity.nodes`` nodeset.
+
+Example::
+
+    len(Entity.nodes)
+
+When we call ``len(Entity.nodes)``, Neomodel will generate a cypher query that counts
+all the nodes with the label ``Entity``. Then that query is executed in the Neo4j database and we get back the count.
+The cypher query string that is generated by Neomodel behind the scene is::
+
+    MATCH (n:Entity) RETURN COUNT(n)
+
+.. note::
+    We are not retrieving all the nodes from the database and then count them. The actual counting is done by the
+    Neo4j database engine which is faster.
+
+Another example, to get count of all the nodes that exist in the PPGDB database::
+
+    len(Entity.nodes) + len(Officer.nodes) + len(Intermediary.nodes) + len(Address.nodes) + len(Other.nodes)
+
+..  todo::
+    Put screenshot image of the console here.
+
+Fetching nodes
+----------------
+In order to retrieve the nodes, read their properties and relationships, actual cypher query need to be executed by Neomodel.
+This is handled completely by Neomodel and we just need to use its query API.
+
+A call to the NodeSet method ``.all()``, would return all the nodes of a nodeset; nevertheless this would result in a expensive
+query. The reason is that Neomodel will actually try to retrieve all the nodes at once. It is recommended to use ``.all()``
+when the nodeset is small. We can reduce the size by filtering the nodeset as will see in the next section.
+
+It is better to fetch the nodes in batches from a nodeset.
+The NodeSet objects support the same operators for indexing and slicing just like the normal python lists.
+
+To get the first element of the ``Entity.nodes`` nodeset::
+
+    Entity.nodes[0]
+
+By slicing we can get batches of the nodeset, which is convenient for writing code that retrieve the nodes in batches of 10.
+For example to get the fist 10 nodes in a list::
+
+    Entity.nodes[0:10]
+
+.. note::
+    Neomodel will generate and execute cypher query only to retrieve the nodes we are asking for.
+    So we are not actually retrieving all the nodes at once from the database. An example of the a cypher query string

```

Comment:
Here, instead of  `An example of the a cypher` can we change it to `An example of a cypher`?